lib/iris/_constraints.py

```diff
@@ -478,3 +478,63 @@ def _cube_func(self, cube):
 
     def __repr__(self):
         return 'AttributeConstraint(%r)' % self._attributes
+
+
+class NameConstraint(Constraint):
+    """
+    Provides a simple Cube name based :class:`Constraint`, which matches
+    against each of the names provided, which may be either standard name,
+    long name, NetCDF variable name and/or the STASH from the attributes
+    dictionary.
+
+    The name constraint will only succeed if all of the provided names match.
+
+    """
+    def __init__(self, **names):
+        """
+        Example usage::
+
+            iris.NameConstraint(long_name='air temp')
+
+            iris.NameConstraint(standard_name='air_temperature',
+                                STASH=lambda stash: stash.item == 203
+
+            .. note:: Name constraint names are case sensitive.
+
+        """
+        self._names = names
+        super(NameConstraint, self).__init__(cube_func=self._cube_func)
```

Comment:
Since we're in Python 3 now, I believe this can be written as `super().__init__(cube_func=self._cube_func)`.